lib/valueflow.cpp

```diff
@@ -2986,7 +2986,7 @@ static const Token* solveExprValues(const Token* expr, std::list<ValueFlow::Valu
         }
         case '*': {
             transformIntValues(values, [&](MathLib::bigint x) {
-                return x / intval;
+                return (intval == 0) ? 0 : x / intval;
```

Comment:
This is not correct. We should not assume `x` is zero in expressions like `x * 0`. Instead we should should just return `expr` when `intval` is `0` and `expr->str()` is `*`, `/`, or `%`.

Reply:
Actually, I think you can just check for `intvalue` is zero and then `break`:

```cpp
        case '*': {
            if (intvalue == 0)
                break;
            transformIntValues(values, [&](MathLib::bigint x) {
                return x / intval;
            });
            return solveExprValues(binaryTok, values);
        }
```

Reply:
Thanks! I will have a look.